### PR TITLE
Complete MO property tax credit logic

### LINKED ISSUE #1361: MO Property Tax Credit TANF Value in Public Assistance
As part of calculating income eligibility for the MO Property Tax Credit, TANF is added to each tax unit's income. However, TANF is at the SPM unit level, meaning multiple tax units could benefit from the 'same' TANF benefit, meaning we need to distribute that benefit across tax units.

### LINKED ISSUE #1608: MO property tax credit logic is incomplete
Currently the MO property tax credit amount is always zero because the formula logic is incomplete.  Part of the incompleteness has been highlighted in issue #1361.  The main missing piece is the phase-out (or reduction) of the credit as income rises.

### PR DESCRIPTION
**Pull request by martinholmer**
Fixes #1608.  
Also, fixes #1361 by adding a `tanf_person` variable.

Subsequent tests are likely to resolve issue #1285 wrt the MO credit tables.

This PR exactly replicates TAXSIM35 credit phase-out results, which are step-like in the extent of phasing out, but do not do the complex two-dimensional rounding to the nearest dollar of the phased-out credit amounts as in the MO tables.

### CODE REVIEW

File: policyengine_us/variables/gov/states/mo/tax/income/credits/property_tax/mo_ptc_taxunit_eligible.py [hunk 1]
```diff
@@ -14,37 +14,32 @@ class mo_property_tax_credit_demographic_tests(Variable):
...
         )
-
-        # Check for receipt of surviving spouse benefits
-        receives_survivor_benefits = (
-            add(tax_unit, period, ["social_security_survivors"]) > 0
-        )
-
+        # check social security survivor benefits eligibility
+        survivor_ben = add(tax_unit, period, ["social_security_survivors"]) > 0
+        aged_head = age_head >= 60
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/mo/tax/income/credits/property_tax/mo_ptc_taxunit_eligible.py:36]: Could you make 60 a parameter?

Quote from [legal code](https://revisor.mo.gov/main/OneSection.aspx?section=135.010&bid=6435) for posterity 
>or if the claimant has reached the age of sixty on or before the last day of the calendar year

File: policyengine_us/variables/gov/states/mo/tax/income/credits/property_tax/mo_ptc_gross_income.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        # compute veterans benefits included in PTC gross income
+        veterans_benefits = person("veterans_benefits", period)
+        exclude = person("is_fully_disabled_service_connected_veteran", period)
+        included_veterans_benefits = ~exclude * veterans_benefits
+        veterans_benefits_income = tax_unit.sum(
+            # Form MO-PTS, line 5
+            included_veterans_benefits
+        )
+        # compute person-level public assistance included in PTC gross income
+        public_assistance_income = tax_unit.sum(
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/mo/tax/income/credits/property_tax/mo_ptc_gross_income.py:45]: Could you make this a separate variable based on a list parameter? Optionally break up some of the other pieces, as this formula is getting a bit long.

File: policyengine_us/variables/gov/states/mo/tax/income/credits/property_tax/mo_property_tax_credit.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        ratio = p.property_tax_rent_ratio
+        rent_limit = p.rent_property_tax_limit
+        rent_amount = min_(rent * ratio, rent_limit)
+        # compute maximum (that is, pre-phaseout) credit amount for taxes
+        ptax = add(tax_unit, period, ["real_estate_taxes"])
+        ptax_limit = p.property_tax_limit
+        ptax_amount = min_(ptax, ptax_limit)
+        # combine the rent_amount and ptax_amount subject to ptax_limit
+        max_credit = min_(rent_amount + ptax_amount, ptax_limit)
+        # phase out credit amount using legislative formula (not form table)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/mo/tax/income/credits/property_tax/mo_property_tax_credit.py:31]: Have you compared this against the table? This was the challenge @nmrodelo and I faced in #1285

**Review by MaxGhenis — COMMENTED** [review]
Thanks @martinholmer! If this also fixes #1361, as it appears to do, could you tag that as well? Same goes for #1285, though we might want some tests against the table (maybe with larger tolerances) to close that one.

File: policyengine_us/variables/gov/states/mo/tax/income/credits/property_tax/mo_property_tax_credit.py (hunk 3 above)
Comment (martinholmer) [reply policyengine_us/variables/gov/states/mo/tax/income/credits/property_tax/mo_property_tax_credit.py:31]: The PolicyEngineUS results are close to the table (which has complex two-dimensional rounding) and are exactly the same as TAXSIM35 generates.

Comment (MaxGhenis) [reply policyengine_us/variables/gov/states/mo/tax/income/credits/property_tax/mo_property_tax_credit.py:31]: Amazing!

File: policyengine_us/parameters/gov/states/mo/tax/income/credits/property_tax/rent_property_tax_limit.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+description: Maximum claimable property tax associated with a rented property.
+values:
+  2008-08-28: 750
```
[the remark this answers is not in the document]
Comment (martinholmer) [reply policyengine_us/parameters/gov/states/mo/tax/income/credits/property_tax/rent_property_tax_limit.yaml:3]: Done.

File: policyengine_us/tests/policy/baseline/gov/states/mo/tax/income/credits/property_tax/mo_ptc_gross_income.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+- name: Unit test for MO property tax credit gross income
```
[the remark this answers is not in the document]
Comment (martinholmer) [reply policyengine_us/tests/policy/baseline/gov/states/mo/tax/income/credits/property_tax/mo_ptc_gross_income.yaml]: Done

File: policyengine_us/parameters/gov/states/mo/tax/income/credits/property_tax/joint_renter_income_offset.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+description: Gross income offset for a married couple filing jointly who rent.
+values:
+  2021-01-01: 2_000
+metadata:
+  unit: currency-USD
+  period: year
+  name: mo_ptc_joint_renter_income_offset
```
[the remark this answers is not in the document]
Comment (martinholmer) [reply policyengine_us/parameters/gov/states/mo/tax/income/credits/property_tax/joint_renter_income_offset.yaml]: Done

**Comment by martinholmer** [comment]
@MaxGhenis said in the discussion of PR #1609:
> Thanks @martinholmer! If this also fixes #1361, as it appears to do, could you tag that as well? 

I've done that.

@MaxGhenis also said:
> Same goes for #1285, though we might want some tests against the table (maybe with larger tolerances) to close that one

Why don't you ask @nmrodelo to add those table tests after this PR is merged?

**Review by MaxGhenis — COMMENTED** [review]
One more nit - could you please organize some of the parameters into subfolders? e.g. `phaseout_rate` and `phaseout_step` into `phase_out.rate` and `phase_out.step`? (We also use `phase_out` in general.)

File: policyengine_us/variables/gov/states/mo/tax/income/credits/property_tax/mo_ptc_taxunit_eligible.py (hunk 1 above)
Comment (martinholmer) [reply policyengine_us/variables/gov/states/mo/tax/income/credits/property_tax/mo_ptc_taxunit_eligible.py]: Done

File: policyengine_us/variables/gov/states/mo/tax/income/credits/property_tax/mo_ptc_gross_income.py (hunk 2 above)
Comment (martinholmer) [reply policyengine_us/variables/gov/states/mo/tax/income/credits/property_tax/mo_ptc_gross_income.py:38]: Done

**Comment by martinholmer** [comment]
@MaxGhenis said in PR #1609:
 > One more nit - could you please organize some of the parameters into subfolders? e.g. phaseout_rate and phaseout_step into phase_out.rate and phase_out.step? (We also use phase_out in general.)

Done.

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**